qcs/src/qpu/execution.rs

```diff
@@ -90,11 +92,16 @@ impl<'a> Execution<'a> {
 
         let native_quil = if compile_with_quilc {
             trace!("Converting to Native Quil");
-            quilc::compile_program(quil, &isa, config)
-                .wrap_err("When attempting to compile your program to Native Quil")?
+            let thread_config = config.clone();
+            spawn_blocking(move || {
+                quilc::compile_program(&quil, &isa, &thread_config)
+                    .wrap_err("When attempting to compile your program to Native Quil")
+            })
+            .await
+            .map_err(|_| eyre!("Error in quilc thread."))??
```

Comment:
nit: `s/thread/task`, just to be consistent with the underlying runtime.

Reply:
```suggestion
            .map_err(|_| eyre!("Task running quilc did not complete."))??
```